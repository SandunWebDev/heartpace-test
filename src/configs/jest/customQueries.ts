import {
	queryHelpers,
	buildQueries,
	Matcher,
	MatcherOptions,
} from '@testing-library/react';

// Custom query to directly match Cypress "data-cy" elements.
const queryAllByDataCy = (
	container: HTMLElement,
	id: Matcher,
	options?: MatcherOptions,
) => queryHelpers.queryAllByAttribute('data-testid', container, id, options);

const getMultipleError = (_c: Element | null, dataCyValue: string) =>
	`Found multiple elements with the data-test attribute of: ${dataCyValue}`;
const getMissingError = (_c: Element | null, dataCyValue: string) =>
	`Unable to find an element with the data-cy attribute of: ${dataCyValue}`;

const [
	queryByDataCy,
	getAllByDataCy,
	getByDataCy,
	findAllByDataCy,
	findByDataCy,
] = buildQueries(queryAllByDataCy, getMultipleError, getMissingError);

export {
	queryByDataCy,
	queryAllByDataCy,
	getByDataCy,
	getAllByDataCy,
	findAllByDataCy,
	findByDataCy,
};
