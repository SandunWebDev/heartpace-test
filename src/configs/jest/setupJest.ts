import '@testing-library/jest-dom';
import ResizeObserver from 'resize-observer-polyfill';

// Mocking ResizeObserver for "Recharts"
global.ResizeObserver = ResizeObserver;
