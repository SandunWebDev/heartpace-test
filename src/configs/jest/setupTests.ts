import '@testing-library/jest-dom';

// Polyfilling ResizeObserver. Currently mainly needed for "Recharts"
import ResizeObserver from 'resize-observer-polyfill';
global.ResizeObserver = ResizeObserver;
