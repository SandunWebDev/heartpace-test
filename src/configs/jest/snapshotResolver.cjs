// Making "snapshot"  files are create besides the test file.

const resolveSnapshotPath = (
	testPath, // Path of the test file being test3ed
	snapshotExtension, // The extension for snapshots (.snap usually)
) => {
	const snapshotFilePath = testPath + snapshotExtension; // Ex. some.test.tsx + '.snap'
	return snapshotFilePath;
};

const resolveTestPath = (
	snapshotFilePath, // The filename of the snapshot (i.e. some.test.tsx.snap)
	snapshotExtension, // The extension for snapshots (.snap)
) => {
	const testPath = snapshotFilePath
		.replace(snapshotExtension, '')
		.replace('__snapshots__/', ''); //Remove the .snap
	return testPath;
};

/* Used to validate resolveTestPath(resolveSnapshotPath( {this} )) */
const testPathForConsistencyCheck = 'some.test.tsx';

module.exports = {
	resolveSnapshotPath,
	resolveTestPath,
	testPathForConsistencyCheck,
};
