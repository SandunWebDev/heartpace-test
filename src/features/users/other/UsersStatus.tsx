import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

export interface UsersStatusProps {
	totalUserAmount: number;
	filteredUserAmount: number;
}

export default function UsersStatus({
	totalUserAmount,
	filteredUserAmount,
}: UsersStatusProps) {
	return (
		<Box
			sx={{
				display: 'inline-block',
				border: '1.5px solid gray',
				padding: '15px 25px',
				borderRadius: '4px',
			}}>
			<Typography gutterBottom color='gray'>
				Filtered/Total Users
			</Typography>
			<Typography variant='h5' component='div' sx={{ fontWeight: 'bold' }}>
				{filteredUserAmount} / {totalUserAmount}
			</Typography>
		</Box>
	);
}
