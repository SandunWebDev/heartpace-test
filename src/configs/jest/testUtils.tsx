import React, { ReactElement } from 'react';
import { render, queries, within, RenderOptions } from '@testing-library/react';
import { Provider as ReduxProvider } from 'react-redux';
import { ToastContainer } from 'react-toastify';

import MuiThemeProvider from '../theme/MuiThemeProvider';
import { store } from '../../services/redux/store';
import * as customQueries from './customQueries';

// All the default providers should goes here. So we don't have to wrap these for each test.
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
	return (
		<MuiThemeProvider>
			<ReduxProvider store={store}>
				<ToastContainer />
				{children}
			</ReduxProvider>
		</MuiThemeProvider>
	);
};

const allQueries = {
	...queries,
	...customQueries,
};

// Re-exporting "React Testing Library's" original "render, screen, within" with our Custom Wrappers and Custom Queries.
const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper' | 'queries'>,
) => render(ui, { wrapper: AllTheProviders, queries: allQueries, ...options });
const customScreen = within(document.body, allQueries);
const customWithin = (element: HTMLElement) => within(element, allQueries);

export * from '@testing-library/react';
export {
	customRender as render,
	customScreen as screen,
	customWithin as within,
};
