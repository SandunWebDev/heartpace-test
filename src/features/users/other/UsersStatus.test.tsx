import { render, screen, cleanup } from '../../../configs/jest/testUtils';
import UsersStatus from './UsersStatus';

const defaultProps = {
	filteredUserAmount: 40,
	totalUserAmount: 100,
};

describe('UsersStatus Component', () => {
	afterEach(cleanup);

	it('snapshot', () => {
		const { asFragment } = render(<UsersStatus {...defaultProps} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render ths given values in "filtered / total" format', () => {
		render(<UsersStatus {...defaultProps} />);

		expect(screen.getByText('Filtered/Total Users')).toBeInTheDocument();
		expect(screen.getByText('40 / 100')).toBeInTheDocument();
	});
});
