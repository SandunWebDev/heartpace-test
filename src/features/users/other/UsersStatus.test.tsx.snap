// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersStatus Component snapshot 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="MuiBox-root css-1qffmkz"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-qrvgcz-MuiTypography-root"
    >
      Filtered/Total Users
    </p>
    <div
      class="MuiTypography-root MuiTypography-h5 css-stnzmu-MuiTypography-root"
    >
      40 / 100
    </div>
  </div>
</DocumentFragment>
`;
